tests/unit/amo/components/TestAddonTitle.js

```diff
@@ -0,0 +1,115 @@
+
+import AddonTitle, { AddonTitleBase } from 'amo/components/AddonTitle';
+import { createInternalAddon } from 'core/reducers/addons';
+import { shallowUntilTarget, fakeAddon, fakeI18n } from 'tests/unit/helpers';
+
+describe(__filename, () => {
+  const render = (props = {}) => {
+    return shallowUntilTarget(
+      <AddonTitle i18n={fakeI18n()} {...props} />,
+      AddonTitleBase,
+    );
+  };
+
+  it('renders a single author', () => {
+    const authorUrl = 'http://olympia.test/en-US/firefox/user/krupa/';
+    const root = render({
+      addon: createInternalAddon({
+        ...fakeAddon,
+        authors: [
+          {
+            name: 'Krupa',
+            url: authorUrl,
+          },
+        ],
+      }),
+    });
+
+    expect(root.html()).toContain('Krupa');
+    expect(root.html()).toContain(authorUrl);
+  });
+
+  it('renders multiple authors', () => {
+    const root = render({
+      addon: createInternalAddon({
+        ...fakeAddon,
+        authors: [
+          {
+            name: 'Krupa',
+            url: 'http://olympia.test/en-US/firefox/user/krupa/',
+          },
+          {
+            name: 'Fligtar',
+            url: 'http://olympia.test/en-US/firefox/user/fligtar/',
+          },
+        ],
+      }),
+    });
+
+    expect(root.html()).toContain('Krupa');
+    expect(root.html()).toContain('Fligtar');
+    expect(root.render().find('a')).toHaveLength(2);
+  });
+
+  it('renders without authors', () => {
+    const addon = createInternalAddon({ ...fakeAddon, authors: null });
+    const root = render({ addon });
+
+    expect(root.html()).toContain(addon.name);
+  });
+
+  it('renders an author without url', () => {
+    const root = render({
+      addon: createInternalAddon({
+        ...fakeAddon,
+        authors: [
+          {
+            name: 'Krupa',
+            url: null,
+          },
+        ],
+      }),
+    });
+
+    expect(root.html()).toContain('Krupa');
+    expect(root.render().find('a')).toHaveLength(0);
+  });
+
+  it('sanitizes a title', () => {
+    const root = render({
+      addon: createInternalAddon({
+        ...fakeAddon,
+        name: '<script>alert(document.cookie);</script>',
+      }),
+    });
+
+    // Make sure an actual script tag was not created.
+    expect(root.find('h1 script')).toHaveLength(0);
+    // Make sure the script removed.
+    expect(root.find('h1').html()).not.toContain('<script>');
+  });
+
+  it('allows certain HTML tags in the title', () => {
+    const name = 'Krupa';
+    const url = 'http://olympia.test/en-US/firefox/user/krupa/';
+
+    const root = render({
+      addon: createInternalAddon({
+        ...fakeAddon,
+        authors: [
+          {
+            name,
+            url,
+          },
+        ],
+      }),
+    });
+
+    // Make sure these tags were whitelisted and make sure the santizer didn't
+    // strip the class attribute:
+    expect(root).toHaveHTML(oneLine`<h1 class="AddonTitle">Chill Out <span
```

Comment:
I will rewrite this test case in the next PR, so I'd like to keep this one as-is for now, given that it was mostly written like that before.

Reply:
What is the next PR? I am actively working on something that will use this (#6616), and I don't want to spend time on something that's just going to have to change again. Will this component change at all in the next PR, or just the tests?

Reply:
@bobsilverberg the interface won't change so that should not be a problem for #6616, unless I am missing something. The internals of this `AddonTitle` component will change a lot, but neither the UI not its API.

Reply:
It's probably better if you use this new component in your PR, so that you'll be able to see if a new class name is really needed.

I filed: https://github.com/mozilla/addons-frontend/issues/6628. I'll open a PR shortly for that, but I can wait until your current PR lands and adapt it.

Reply:
PR is here in case you need an idea of what is supposed to change: https://github.com/mozilla/addons-frontend/pull/6630